exonum/src/blockchain/mod.rs

```diff
@@ -74,6 +75,8 @@ mod tests;
 /// into a single network.
 pub struct Blockchain {
     db: Arc<dyn Database>,
+    last_snapshot: Option<(Height, PathBuf)>,
```

Comment:
Yes, I can move DB image creation out of `Blockchain` into separate module and to make backups asynchronously, I can start a separate thread that will watch blockchain state and will make backups on specific heights. 

Another question is, how to prevent data races during snapshot creation if it will be done asynchronously to `Blockchain`.